pkg/activator/throttler_test.go

```diff
@@ -0,0 +1,249 @@
+	throttler.breakers[revID] = queue.NewBreaker(throttler.breakerParams)
+	got := len(throttler.breakers)
+	if got != 1 {
+		t.Errorf("Unexpected number of Breakers was created. Want %d, got %d", 1, got)
+	}
+	throttler.Remove(revID)
+	got = len(throttler.breakers)
+	if got != 0 {
+		t.Errorf("Unexpected number of Breakers was created. Want %d, got %d", 0, got)
+	}
+}
+
+func TestHelper_UpdateEndpoints(t *testing.T) {
+	want := int32(10)
+	throttler := getThrottler(defaultMaxConcurrency, existingRevisionGetter(10), existingEndpointsGetter, t)
+	throttler.breakers[revID] = queue.NewBreaker(throttler.breakerParams)
+	updater := UpdateEndpoints(throttler)
+
+	endpointsBefore := corev1.Endpoints{ObjectMeta: v1.ObjectMeta{Name: revID.Name + "-service", Namespace: revID.Namespace}, Subsets: testinghelper.GetTestEndpointsSubset(0, 1)}
+	endpointsAfter := corev1.Endpoints{ObjectMeta: v1.ObjectMeta{Name: revID.Name + "-service", Namespace: revID.Namespace}, Subsets: testinghelper.GetTestEndpointsSubset(1, 1)}
+	updater(&endpointsBefore, &endpointsAfter)
+
+	breaker, _ := throttler.breakers[revID]
+	got := breaker.Capacity()
+	if got != want {
+		t.Errorf("Unexpected Breaker capacity received. Want %d, got %d", want, got)
+	}
+}
+
+func TestHelper_UpdateEndpoints_WithDeltaMoreThenOne(t *testing.T) {
+	want := int32(20)
+	throttler := getThrottler(int32(20), existingRevisionGetter(10), existingEndpointsGetter, t)
+	throttler.breakers[revID] = queue.NewBreaker(throttler.breakerParams)
+	updater := UpdateEndpoints(throttler)
+
+	endpointsBefore := corev1.Endpoints{ObjectMeta: v1.ObjectMeta{Name: revID.Name + "-service", Namespace: revID.Namespace}, Subsets: testinghelper.GetTestEndpointsSubset(0, 1)}
+	endpointsAfter := corev1.Endpoints{ObjectMeta: v1.ObjectMeta{Name: revID.Name + "-service", Namespace: revID.Namespace}, Subsets: testinghelper.GetTestEndpointsSubset(2, 1)}
+	updater(&endpointsBefore, &endpointsAfter)
+
+	breaker, _ := throttler.breakers[revID]
+	got := breaker.Capacity()
+	if got != want {
+		t.Errorf("Unexpected Breaker capacity received. Want %d, got %d", want, got)
+	}
+}
+
+func TestHelper_DeleteBreaker(t *testing.T) {
+	throttler := getThrottler(int32(20), existingRevisionGetter(10), existingEndpointsGetter, t)
+	revision := &v1alpha1.Revision{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      revID.Name,
+			Namespace: revID.Namespace,
+		},
+	}
+	revID := RevisionID{Namespace: revID.Namespace, Name: revID.Name}
+	throttler.breakers[revID] = queue.NewBreaker(throttler.breakerParams)
+	if len(throttler.breakers) != 1 {
+		t.Errorf("Breaker map size didn't change. Wanted %d, got %d", 1, len(throttler.breakers))
+	}
+	deleter := DeleteBreaker(throttler)
+	deleter(revision)
+	if len(throttler.breakers) != 0 {
+		t.Error("Breaker map is not empty")
+	}
+}
+
+func getThrottler(maxConcurrency int32, revisionGetter func(RevisionID) (*v1alpha12.Revision, bool, error), endpointsGetter func(RevisionID) (int32, error), t *testing.T) *Throttler {
```

Comment:
Golang doesn't really like passing `t` around. Can you pass in the logger directly here to avoid passing `t`?